### PR TITLE
WIP Refactor/typescript

### PR DESCRIPTION
**Pull request by camacho**
This PR is for refactoring the project to use TypeScript. 

Additional scripts, tools, and dependencies were added to support this change.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ aliases:
     - checkout
     - attach_workspace:
       at: *working-directory
+    - steps:
+        - run: yarn install
```
Comment (camacho) [inline .circleci/config.yml]: Not sure how we got away with not having this. Maybe I'm missing something 🤷‍♂

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,7 +21,8 @@ jobs:
             - yarn-packages-v1-{{ .Branch }}-{{ checksum "yarn.lock" }}
             - yarn-packages-v1-{{ .Branch }}-
             - yarn-packages-v1-
-      - run: 'npx greenkeeper-lockfile@next greenkeeper-lockfile-upload'
+      - run: npx greenkeeper-lockfile@next greenkeeper-lockfile-upload
+      - run: yarn install --no-progress --silent
```
Comment (camacho) [inline .circleci/config.yml]: Not sure how we got away with not having this. Maybe I'm missing something 🤷‍♂

File: .gitignore [hunk 3]
```diff
@@ -54,7 +57,9 @@ typings/
 # Yarn Integrity file
 .yarn-integrity
 
-# dotenv environment variables file
+# Dotenv environment variables file
```
Comment (camacho) [inline .gitignore]:
```suggestion
# dotenv environment variables file
```

**Comment by camacho** [comment]
Closing in favor of #64

File: .circleci/config.yml (hunk 2 above)
Comment (peterkc) [reply .circleci/config.yml]: Oh. Maybe it was the caching of the modules? Good catch.

**Pull request closed without merging**